redo svg filters, match font stack

SArpnt opened the pull request:
this fixes blur in firefox (especially on text), gives the regular snap style much more accurate block borders and text shadow, and removes quite a bit of bad code.

there's several unrelated changes in these commits so feel free to cherry pick if one of the commits is real bad

before:
![image](https://github.com/user-attachments/assets/46b74411-f956-4e47-bc2f-54c2a6832429)
snap:
![image](https://github.com/user-attachments/assets/7e64a56f-ce2d-4455-adda-4c708620cb4b)
after:
![image](https://github.com/user-attachments/assets/2f73c03e-19e2-4a61-97c4-ea21d4bba463)

SArpnt committed:
use same font stack as snap

SArpnt committed:
simpler scratch 2 dark/light filter

fits in better with snap block changes in later commits

SArpnt committed:
simpler scratch 3 zebra filter

fits in better with snap block changes in later commits

SArpnt committed:
better scratch 2 bevel filter

shadow edges aren't clipped twice.
i don't think the old filter would blur in firefox but the general method used usually would.
this new method uses patterns that shouldn't have that happen.

this fits in better with snap block changes in later commits.

SArpnt committed:
remove all copies of filter.js

they contain serious bugs, make inefficient output, usually cause blur on firefox,
and aren't much more ergonomic than writing out the filter manually.

ego-lay-atman-bay commented:
After testing it out, I can see that it's now worse in chrome.
![image](https://github.com/user-attachments/assets/ff298f2f-503b-4992-8bb1-1c92e6c2d4be)

However it does look better in firefox
![image](https://github.com/user-attachments/assets/6c73ed99-84ec-4961-9c2c-c829813b3639)

When I zoom in on both chrome and firefox, they do look the same, so it's just a matter of how each engine scales svgs. I'm not going to merge this right now, since it does give a worse look on chrome now.

Perhaps the better thing to do about the bevels is actually changing them to how snap draws them, as paths with a linear gradient. Sure, it'll bloat the svgs even more, but it might make it work better on both browser engines.

SArpnt commented:
hm, i was testing changes on chrome when starting out but assumed the later changes would be fine. it's all out of order now after all the rebasing. i should still be able to make chrome match firefox with some small tweaks.

SArpnt committed:
fix blurry drop shadow text in firefox, better drop shadow at normal scale

previously filter bounds and particular effects (mostly feFlood)
would cause some problems with element dimensions in firefox and make
that element blurry and incorrectly positioned. many filters also
had coordinates rounded differently between the browsers.

this new svg filter act the same between browsers and
is nicer and simpler in several ways

SArpnt committed:
overhaul snap block bevel filter

similar to the previous commit, the block bevel caused blur and misalignment in firefox.
this new filter is also more accurate in general due to other tweaks.
it looks slightly different between browsers but it's close enough.

SArpnt commented:
it looks almost exactly the same in both browsers to me?

firefox 100% zoom:
![image](https://github.com/user-attachments/assets/5b6f77e5-447c-458b-abd2-7ccbc777d26b)
chrome 100% zoom:
![image](https://github.com/user-attachments/assets/8810e286-a217-4e3c-90a0-ad70145d59e0)
firefox 200% zoom:
![image](https://github.com/user-attachments/assets/9d33adb2-0137-4138-9e88-2398d01f10e8)
chrome 200% zoom:
![image](https://github.com/user-attachments/assets/19122050-ce97-443a-bb14-e3d7ba895e78)

chrome has a harder border at 100% zoom and the fat borders at larger zooms isn't great (i probably turned the blur down too low), but i'm not getting what you have at all.
you implied that what you're getting now is when it isn't zoomed in. do you have a high dpi screen? if that's changing how it scales i don't know how i can test for that.

SArpnt commented:
i adjusted the bevel edges to add a slight bit more blur for when zoomed in (i don't think i can get away with more without messing up normal sizes) and reordered the commits from least to most invasive changes.

you should be able to pull up to the blurry text fix for now if the bevel edges are still an issue. i think the text change is the most important fix out of the set.

ego-lay-atman-bay commented:
It might be a DPI thing, since I have my display scale set to 125%. When I change it to 100% chrome looks the same as firefox. Also, it still looks the same as it did before (or at least, I currently don't see any changes).

100% Display scale:
Chrome 100% zoom:
![image](https://github.com/user-attachments/assets/257659d9-3d11-4ef0-8317-35888ff5bb3a)

Firefox 100% zoom:
![image](https://github.com/user-attachments/assets/71ee3b7f-3b38-42e3-a95f-67648e093c8f)

Chrome 200% zoom:
![image](https://github.com/user-attachments/assets/d62b056d-2a18-4022-871e-e715d6fc3d0b)

Firefox 200% zoom:
![image](https://github.com/user-attachments/assets/392877fa-fbc8-4d8a-943c-48646faca605)

125% Display scale:
Chrome 100% zoom:
![image](https://github.com/user-attachments/assets/44dbd071-990e-4a63-82c1-ef2f17b82630)

Firefox 100% zoom:
![image](https://github.com/user-attachments/assets/fb86a4a2-9a7a-4119-8770-6cf452087a98)

Chrome 200% zoom:
![image](https://github.com/user-attachments/assets/1e1aab87-768a-4988-9275-f77469b8a86e)

Firefox 200% zoom:
![image](https://github.com/user-attachments/assets/77021889-c3e1-477a-85f2-ef2bd5fa17e3)

SArpnt commented:
i only made a very subtle change to the block's lighting. there's really not a lot of room for adjustment.

maybe a convolution matrix could shift the text over by a physical pixel more reliably? it would probably be much simpler to just generate more svg nodes. give the text an id and \<use\> it for the shadow. i don't think more work on the svg filters is worth it.

if/when you go down the route of making thousands of similar paths i'd recommend you just put each block once in \<defs\> and then reuse them as [border images](https://developer.mozilla.org/en-US/docs/Web/CSS/border-image). i haven't checked if svg widely supports border images but if not you can simulate it manually with many \<use\>s and transforms. i've been using border images to render blocks with dead simple html for my own wip project.
